[SPARK-14861][SQL] Replace internal usages of SQLContext with SparkSession

andrewor14 opened the pull request:
## What changes were proposed in this pull request?

In Spark 2.0, `SparkSession` is the new thing. Internally we should stop using `SQLContext` everywhere since that's supposed to be not the main user-facing API anymore.

In this patch I took care to not break any public APIs. The one place that's suspect is `o.a.s.ml.source.libsvm.DefaultSource`, but according to @mengxr it's not supposed to be public so it's OK to change the underlying `FileFormat` trait.

**Reviewers**: This is a big patch that may be difficult to review but the changes are actually really straightforward. If you prefer I can break it up into a few smaller patches, but it will delay the progress of this issue a little.
## How was this patch tested?

No change in functionality intended.

Commit:
Refactor Dataset and related APIs

Commit:
Refactor SQL stream related classes

Commit:
Refactor QueryExecution

Commit:
Refactor RunnableCommand

Commit:
Refactor DataSource

Commit:
Refactor SessionState

Commit:
Refactor HiveMetastoreCatalog

Commit:
Refactor HiveStrategies

Commit:
Refactor FileFormat

Commit:
Fix style

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

SparkQA commented:
**[Test build #56747 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56747/consoleFull)** for PR 12625 at commit [`d1a6592`](https://github.com/apache/spark/commit/d1a6592f8cf2696541f422481a2ee16c019e0ed0).

andrewor14 commented on mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala:
```diff
@@ -356,14 +356,14 @@ private[clustering] trait LDAParams extends Params with HasFeaturesCol with HasM
     @Since("1.6.0") override val uid: String,
     @Since("1.6.0") val vocabSize: Int,
-    @Since("1.6.0") @transient protected val sqlContext: SQLContext)
+    @Since("1.6.0") @transient protected[ml] val sparkSession: SparkSession)
```
this will break MIMA but it's probably OK because we don't expect people to extend `LDAModel`

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/ContinuousQueryManager.scala:
```diff
@@ -29,16 +29,16 @@ import org.apache.spark.sql.util.ContinuousQueryListener
  */
 @Experimental
-class ContinuousQueryManager(sqlContext: SQLContext) {
+class ContinuousQueryManager(sparkSession: SparkSession) {
```
this is a public API, but it's not in any release yet so it's probably OK to change it.

SparkQA commented:
**[Test build #56747 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56747/consoleFull)** for PR 12625 at commit [`d1a6592`](https://github.com/apache/spark/commit/d1a6592f8cf2696541f422481a2ee16c019e0ed0).
 * This patch **fails MiMa tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `case class OptimizeIn(conf: CatalystConf) extends Rule[LogicalPlan] `

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56747/
Test FAILed.

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala:
```diff
@@ -58,7 +58,7 @@ import org.apache.spark.sql.util.ExecutionListenerManager
  */
 class SQLContext private[sql](
-    @transient private val sparkSession: SparkSession,
+    @transient val sparkSession: SparkSession,
```
Note: new public API

Commit:
Fix MIMA + minor fixes

SparkQA commented:
**[Test build #56750 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56750/consoleFull)** for PR 12625 at commit [`148ed3e`](https://github.com/apache/spark/commit/148ed3e8a6a381e875658866f7ce9ca7fc279058).

SparkQA commented:
**[Test build #56750 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56750/consoleFull)** for PR 12625 at commit [`148ed3e`](https://github.com/apache/spark/commit/148ed3e8a6a381e875658866f7ce9ca7fc279058).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `sealed abstract class LDAModel protected[ml] (`
  * `class LocalLDAModel protected[ml] (`
  * `class DistributedLDAModel protected[ml] (`
  * `class DataFrameReader protected[sql](sparkSession: SparkSession) extends Logging `
  * `class Dataset[T] protected[sql](`

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56750/
Test FAILed.

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

Commit:
Fix tests

It turns out that MLlib expects SQLContext's to be serializable,
so SparkSession must also be serializable.

SparkQA commented:
**[Test build #56910 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56910/consoleFull)** for PR 12625 at commit [`2c88b4e`](https://github.com/apache/spark/commit/2c88b4ea6a84d2966d6945103e9aac353a3dee54).

Commit:
Fix Hive tests compile

SparkQA commented:
**[Test build #56911 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56911/consoleFull)** for PR 12625 at commit [`521292e`](https://github.com/apache/spark/commit/521292e4ad66dbdcbd90f1e083238a3561a816cc).

SparkQA commented:
**[Test build #56910 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56910/consoleFull)** for PR 12625 at commit [`2c88b4e`](https://github.com/apache/spark/commit/2c88b4ea6a84d2966d6945103e9aac353a3dee54).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56910/
Test FAILed.

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

SparkQA commented:
**[Test build #56920 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56920/consoleFull)** for PR 12625 at commit [`9b78c04`](https://github.com/apache/spark/commit/9b78c04f801a393e97abb3919ed6228671d83681).

SparkQA commented:
**[Test build #56911 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56911/consoleFull)** for PR 12625 at commit [`521292e`](https://github.com/apache/spark/commit/521292e4ad66dbdcbd90f1e083238a3561a816cc).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56911/
Test FAILed.

SparkQA commented:
**[Test build #56920 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56920/consoleFull)** for PR 12625 at commit [`9b78c04`](https://github.com/apache/spark/commit/9b78c04f801a393e97abb3919ed6228671d83681).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56920/
Test FAILed.

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

Commit:
Fix test

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

SparkQA commented:
**[Test build #56931 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56931/consoleFull)** for PR 12625 at commit [`9652951`](https://github.com/apache/spark/commit/9652951f0e32bad0ab872135fb87294ff00fbeda).

SparkQA commented:
**[Test build #56932 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56932/consoleFull)** for PR 12625 at commit [`4c427c0`](https://github.com/apache/spark/commit/4c427c0c284d4350c9495b748a00e90fea6cad5d).

SparkQA commented:
**[Test build #2875 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2875/consoleFull)** for PR 12625 at commit [`4c427c0`](https://github.com/apache/spark/commit/4c427c0c284d4350c9495b748a00e90fea6cad5d).

SparkQA commented:
**[Test build #56932 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56932/consoleFull)** for PR 12625 at commit [`4c427c0`](https://github.com/apache/spark/commit/4c427c0c284d4350c9495b748a00e90fea6cad5d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56932/
Test PASSed.

SparkQA commented:
**[Test build #56931 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56931/consoleFull)** for PR 12625 at commit [`9652951`](https://github.com/apache/spark/commit/9652951f0e32bad0ab872135fb87294ff00fbeda).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56931/
Test PASSed.

SparkQA commented:
**[Test build #2875 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2875/consoleFull)** for PR 12625 at commit [`4c427c0`](https://github.com/apache/spark/commit/4c427c0c284d4350c9495b748a00e90fea6cad5d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

Commit:
Merge branch 'master' of github.com:apache/spark into spark-session-refactor

SparkQA commented:
**[Test build #2880 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2880/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).

SparkQA commented:
**[Test build #2881 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2881/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).

SparkQA commented:
**[Test build #2882 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2882/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).

SparkQA commented:
**[Test build #56950 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56950/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).

SparkQA commented:
**[Test build #2881 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2881/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class UnresolvedException[TreeType <: TreeNode[_]](tree: TreeType, function: String)`
  * `case class UnresolvedGenerator(name: FunctionIdentifier, children: Seq[Expression])`
  * `class RuntimeConfigImpl(`

SparkQA commented:
**[Test build #2882 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2882/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class UnresolvedException[TreeType <: TreeNode[_]](tree: TreeType, function: String)`
  * `case class UnresolvedGenerator(name: FunctionIdentifier, children: Seq[Expression])`
  * `class RuntimeConfigImpl(`

SparkQA commented:
**[Test build #2880 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2880/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class UnresolvedException[TreeType <: TreeNode[_]](tree: TreeType, function: String)`
  * `case class UnresolvedGenerator(name: FunctionIdentifier, children: Seq[Expression])`
  * `class RuntimeConfigImpl(`

rxin commented:
Merging in master.

SparkQA commented:
**[Test build #56950 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56950/consoleFull)** for PR 12625 at commit [`e47fbf0`](https://github.com/apache/spark/commit/e47fbf0de63e78dfdc6b16b1d844dacb2aa09f68).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class UnresolvedException[TreeType <: TreeNode[_]](tree: TreeType, function: String)`
  * `case class UnresolvedGenerator(name: FunctionIdentifier, children: Seq[Expression])`
  * `class RuntimeConfigImpl(`

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56950/
Test PASSed.
